flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/jsonplan/JsonPlanGenerator.java

```diff
@@ -117,10 +118,27 @@ public static String generatePlan(
                 // write the core properties
                 JobVertexID vertexID = vertex.getID();
                 int storeParallelism = vertexParallelism.getParallelism(vertexID);
+                int parallelism =
+                        storeParallelism != -1 ? storeParallelism : vertex.getParallelism();
                 gen.writeStringField("id", vertexID.toString());
-                gen.writeNumberField(
-                        "parallelism",
-                        storeParallelism != -1 ? storeParallelism : vertex.getParallelism());
+                gen.writeNumberField("parallelism", parallelism);
+                int storeMaxParallelism =
+                        vertexParallelism.getMaxParallelismForVertices() == null
+                                ? -1
+                                : vertexParallelism
+                                        .getMaxParallelismForVertices()
+                                        .getOrDefault(vertexID, -1);
+                int maxParallelism =
+                        storeMaxParallelism == -1
```

Comment:
Too many `-1` here, we should use `JobVertex#MAX_PARALLELISM_DEFAULT` to improve human readability.

Reply:
ok